extensions/amp-video/0.1/mixins.js

```diff
@@ -0,0 +1,524 @@
+   * @param {function(boolean)} setMuted
+   */
+  installEventHandlers(eventsToForward, setMuted) {
+    // Merely keep track to install once state is transfered.
+    this.eventsToForward_ = eventsToForward;
+    this.setMuted_ = setMuted;
+  }
+
+  /** @param {!PlaybackMixin} mixin */
+  transferState(mixin) {
+    const dummyElement = dev().assertElement(this.dummyElement_);
+    const {element} = this.impl_;
+    const {attributes} = dummyElement;
+    const {classList} = dummyElement;
+    const baseNode = mixin.getBaseElement();
+    const fragment =
+        dev().assert(element.ownerDocument).createDocumentFragment();
+
+    copyChildren(dummyElement, baseNode);
+
+    for (let i = 0; i < attributes.length; i++) {
+      const {name, value} = attributes.item(i);
+      baseNode.setAttribute(name, value);
+    }
+
+    for (let i = 0; i < classList.length; i++) {
+      const className = classList.item(i);
+      baseNode.classList.add(className);
+    }
+
+    mixin.installEventHandlers(
+        /** @type {!Array<string>} */ (dev().assert(this.eventsToForward_)),
+        /** @type {function(boolean)} */ (dev().assert(this.setMuted_)));
+
+    baseNode.appendChild(fragment);
+
+    this.impl_ = null; // GC
+    this.dummyElement_ = null; // GC
+    this.eventsToForward_ = null; // GC
+    this.setMuted_ = null; // GC
+
+    return mixin;
+  }
+
+  /** @final */
+  isIntermediate() {
+    return true;
+  }
+
+  /** @final */
+  controlsPlayback() {
+    return false;
+  }
+}
+
+
+export class VideoElementMixin extends PlaybackMixin {
+
+  /** @param {!./amp-video.AmpVideo} impl */
+  constructor(impl) {
+    super(impl);
+
+    /** @protected {?HTMLMediaElement} */
+    // TODO(alanorozco): Make private
+    this.video = null;
+  }
+
+  isVideoSupported() {
+    return !!this.video.play;
+  }
+
+  /** @override */
+  getBaseElement() {
+    if (this.video) {
+      return this.video;
+    }
+
+    const {element} = this.impl;
+
+    this.video = /** @type {!HTMLMediaElement} */ (
+      dev().assert(element.ownerDocument).createElement('video'));
+
+    this.impl.getVsync().mutate(() => {
+      element.appendChild(this.video);
+    });
+
+    return this.video;
+  }
+
+  /** @override */
+  installEventHandlers(eventsToForward, setMuted) {
+    const video = dev().assertElement(this.video);
+
+    this.impl.forwardEvents(eventsToForward, video);
+
+    listen(video, 'volumechange', () => {
+      setMuted(video.muted);
+    });
+  }
+
+  /** @override */
+  whenLoadStarts() {
+    // loadPromise for media elements listens to `loadstart`
+    return this.impl.loadPromise(this.video);
+  }
+
+  /** @override */
+  play() {
+    new Promise(() => this.video.play()).catch(() => {
+      // Empty catch to prevent useless unhandled promise rejection logging.
+      // Play can fail for many reasons such as video getting paused before
+      // play() is finished.
+      // We use events to know the state of the video and do not care about
```

Comment:
"We" meaning "amp-video"?  Are you sure 0p consumers of amp-video don't rely on this?